src/overlays/actors/ovl_En_Az/z_en_az.c

```diff
@@ -39,108 +83,1755 @@ const ActorInit En_Az_InitVars = {
+};
+static Vec3f D_80A993AC[3] = {
+    { 1.0f, 1.0f, 0.0f },
+    { 0.9f, 0.9f, 120.0f },
+    { 0.95f, 0.95f, 240.0f },
+};
+static Vec3f D_80A993D0[3] = {
+    { 1.5f, 1.5f, 0.0f },
+    { 1.2f, 1.2f, 120.0f },
+    { 1.35f, 1.35f, 240.0f },
+};
+static TexturePtr D_80A993F4[4] = {
+    object_az_Tex_00F918,
+    object_az_Tex_010118,
+    object_az_Tex_010918,
+    object_az_Tex_011118,
+};
+static TexturePtr D_80A99404[3] = {
+    object_az_Tex_016018,
+    object_az_Tex_016818,
+    object_az_Tex_017018,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A97D5C.s")
+void EnAz_Draw(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    EnAz* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A97E48.s")
+    OPEN_DISPS(play->state.gfxCtx);
+    POLY_OPA_DISP = Gfx_CallSetupDL(POLY_OPA_DISP, 0x19);
+    CLOSE_DISPS(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A97EAC.s")
+    if (this->unk_374 & 2) {
+        SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                              func_80A98DA4, func_80A98E48, &this->actor);
+    } else {
+        OPEN_DISPS(play->state.gfxCtx);
+        gSPSegment(POLY_OPA_DISP++, 8, Lib_SegmentedToVirtual(D_80A993F4[this->unk_37E]));
+        gSPSegment(POLY_OPA_DISP++, 9, Lib_SegmentedToVirtual(D_80A99404[this->unk_380]));
+        SkelAnime_DrawFlexOpa(play, this->skelAnime.skeleton, this->skelAnime.jointTable, this->skelAnime.dListCount,
+                              func_80A98DA4, func_80A98E48, &this->actor);
+        CLOSE_DISPS(play->state.gfxCtx);
+    }
+    OPEN_DISPS(play->state.gfxCtx);
+    if ((this->actor.depthInWater >= 28.0f) && (this->actor.speedXZ > 0.5f)) {
+        Matrix_Translate(this->unk_3B4.x, this->unk_3B4.y, this->unk_3B4.z, 0);
+        Matrix_RotateYS(this->actor.shape.rot.y, 1);
+        Matrix_RotateXS(this->actor.shape.rot.x, 1);
+        Matrix_RotateZS(this->actor.shape.rot.z, 1);
+        Matrix_RotateXS(this->unk_39E, 1);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, 1);
+        func_8012C2DC(play->state.gfxCtx);
+        if (this->unk_374 & 2) {
+            s32 i;
+            Vec3f sp98;
+            f32 sp94;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A97F9C.s")
+            Matrix_Translate(0.0f, 0.0f, -2000.0f, 1);
+            for (i = 0; i < 5; i++) {
+                Matrix_Push();
+                func_80A98F94(D_80A9930C[i], this->unk_39C, &sp94);
+                AnimatedMat_Draw(play, Lib_SegmentedToVirtual(D_80A99180[i]));
+                Matrix_Translate(0.0f, sp94 * 100.0f, i * -930.0f, 1);
+                Matrix_Scale(1.1f, 0.95f, 1.0f, 1);
+                func_80124618(D_80A991A4[i], this->unk_39C, &sp98);
+                Matrix_Scale(sp98.x, sp98.y, sp98.z, 1);
```

Comment:
```suggestion
                Matrix_Translate(0.0f, sp94 * 100.0f, i * -930.0f, MTXMODE_APPLY);
                Matrix_Scale(1.1f, 0.95f, 1.0f, MTXMODE_APPLY);
                func_80124618(D_80A991A4[i], this->unk_39C, &sp98);
                Matrix_Scale(sp98.x, sp98.y, sp98.z, MTXMODE_APPLY);
```
There's a few more of these below, could you fill these out?